hooks/utils.py

```diff
@@ -290,6 +247,10 @@ def install_builtin_server():
     requirements = os.path.join(CURRENT_DIR, 'server-requirements.pip')
     # Install the builtin server dependencies avoiding to download requirements
     # from the network.
+    # XXX frankban: this pip installation here implicitly depends on juju-gui
+    # dependencies to be installed. In essence, this function does not fetch
+    # dependencies from the network only incidentally, because setup_gui() has
+    # been already called in the unit.
```

Comment:
Ok, that makes sense. Thanks.

> On Apr 14 2016, at 8:35 am, Francesco Banconi
&lt;notifications@github.com&gt; wrote:  

>

> In [hooks/utils.py](https://github.com/juju/juju-gui-
charm/pull/59#discussion_r59708660):

>  
>  
>     > @@ -290,6 +247,10 @@ def install_builtin_server():

>     >      requirements = os.path.join(CURRENT_DIR, 'server-
requirements.pip')

>     >      # Install the builtin server dependencies avoiding to download
requirements

>     >      # from the network.

>     > +    # XXX frankban: this pip installation here implicitly depends on
juju-gui

>     > +    # dependencies to be installed. In essence, this function does
not fetch

>     > +    # dependencies from the network only incidentally, because
setup_gui() has

>     > +    # been already called in the unit.

>  
>

> Well, in theory we should fix this, by either including all dependencies in
the two places or by collecting deps in a single place used by both functions.
I still consider this suboptimal, hence the XXX.

>

> —  
You are receiving this because you commented.  
Reply to this email directly or [view it on GitHub](https://github.com/juju
/juju-gui-charm/pull/59/files/92cb8498385672757523062780ae1a85f097128e#r597086
60)![](https://github.com/notifications/beacon/AABoaIggtuvcP0AU3n6gdQMwNeRoUQ6
dks5p3jR1gaJpZM4IHS48.gif)